sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2630,7 +2630,10 @@ object GpuOverrides {
       (shuffle, conf, p, r) => new GpuShuffleMeta(shuffle, conf, p, r)),
     exec[UnionExec](
       "The backend for the union operator",
-      ExecChecks(TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL, TypeSig.all),
+      ExecChecks((TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL + TypeSig.STRUCT
+        .withPsNote(TypeEnum.STRUCT, "unionByName will not optionally fill missing columns with " +
+          "nulls when the col has structs"))
+        .nested(), TypeSig.all),
```

Comment:
do we want to allow this for multi- level nesting right away? In the sort PR we are doing [single-level first](https://github.com/NVIDIA/spark-rapids/pull/1883/files#diff-8926565c5f6bc26203bed459ac37ea0fba15f9f5488aa2412aeac26b31f80ea6R430)

Reply:
Correct me if I am wrong but won't they both result in the same code? 

https://github.com/NVIDIA/spark-rapids/blob/236caa07d353f5455edf97ca158e87920f988123/sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala#L180

https://github.com/NVIDIA/spark-rapids/blob/236caa07d353f5455edf97ca158e87920f988123/sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala#L188

because in your case you are basically passing the `nesting.initialTypes = initialTypes` so how is it different?

Reply:
The difference is whether Struct spills into `nested` types , you can observe the difference in the REPL:
```scala
scala> val pluginSupportedOrderableSig = (
     |     TypeSig.commonCudfTypes +
     |     TypeSig.NULL +
     |     TypeSig.DECIMAL +
     |     TypeSig.STRUCT.nested(
     |       TypeSig.commonCudfTypes +
     |       TypeSig.NULL +
     |       TypeSig.DECIMAL
     |     ))
pluginSupportedOrderableSig: com.nvidia.spark.rapids.TypeSig = com.nvidia.spark.rapids.TypeSig@468f547e

scala> val typeSigBlanketNested = (TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL + TypeSig.STRUCT).nested()

scala> val singleNested = new StructType().add("value", IntegerType)
singleNested: org.apache.spark.sql.types.StructType = StructType(StructField(value,IntegerType,true))

scala> val doubleNested = new StructType().add("nestedInt", new StructType().add("value", IntegerType))
doubleNested: org.apache.spark.sql.types.StructType = StructType(StructField(nestedInt,StructType(StructField(value,IntegerType,true)),true))

scala> pluginSupportedOrderableSig.isSupportedByPlugin(singleNested, true)
res8: Boolean = true

scala> pluginSupportedOrderableSig.isSupportedByPlugin(doubleNested, true)
res7: Boolean = false
```

But with typeSigBlanketNested e.g double nesting would be allowed
```scala
scala> typeSigBlanketNested.isSupportedByPlugin(doubleNested, true)
res11: Boolean = true
````

Reply:
Oh of course! I missed a very important thing that the struct will be a part of the initialTypes. Thanks for the explanation

I have pushed an update. PTAL